src/components/Api.js

```diff
@@ -0,0 +1,162 @@
+export default class Api {
+  constructor(apiSettings) {
+    this._apiSettings = apiSettings;
+    this._headers = apiSettings.headers;
+    this._baseUrl = apiSettings.baseUrl;
+    this._userInfoUrl = `${this._baseUrl}/users/me`;
+    this._cardsUrl = `${this._baseUrl}/cards`;
+    this._avatarUrl = `${this._userInfoUrl}/avatar`;
+  }
+
+  _checkResponse(response) {
+    if (!response.ok) {
+      throw response;
+    }
+  }
+
+  async getInitialCards() {
+    try {
+      const res = await fetch(this._cardsUrl, {
+        headers: this._headers,
+      });
+      this._checkResponse(res);
+      const cardsArray = await res.json();
+      console.log(cardsArray);
+      return cardsArray;
```

Comment:
`await res.json()` should be inside `_checkResponse` according to the task not to duplicate the code


```js
      return  await this._checkResponse(res);


```